### DISCUSSION TITLE
What is the Cuda and Pytorch version that is required to launch reForge WebUI?

### DISCUSSION THREAD

**Discussion by WardCraft** [Panchovix/stable-diffusion-webui-reForge#192] [Q&A]
I'm pretty noob here but I  cannot find it anywhere in Git for this package requirement...

So I have my NVidia 3070 RTX with updated Driver Version: 566.14 and CUDA Version: 12.7  on my system and when I launch reForge WebUI through Stability Matrix, I got a following error:

<img width="752" alt="stability matrix reForge Webui Packages" src="https://github.com/user-attachments/assets/4c04477a-552a-4339-bf02-a3638f2325a9">

Of course I could resolve it by using the command line  --skip-torch-cuda-test   but I don't intend to.   I want to run on my graphic with cuda for  optimization.

**Comment by Deaquay** [comment]
Don't know exactly what is required, but this should work for you as well.

version: f1.1.5-v1.10.1RC-latest-876-g62d901b3  •  python: 3.10.6  •  torch: 2.5.1+cu124  •  xformers: 0.0.29  •  gradio: 3.41.2  •  checkpoint: e96ec818fa

![image](https://github.com/user-attachments/assets/f4d99c8e-bc3d-4175-a090-8d1b4ef60472)

**Comment by pycms-nube** [comment]
The problem somehow related to stability matrix itself, the new update include a function to select the desired GPU to run any instance. 
For your GPU, you might need a cu124 torch, go to the “Python Package” to see what version of PyTorch it’s been installed, especially Windows platform, check if it comes with +cu124 ending. If it’s come with cu124 ending, the problem might be related to the desired GPU has been set to wrong GPU.  Linux platform does not need cu124 ending, but need to check what CUDA version is currently installed.
